JustTrack/JEEventsGenerator/main.swift

```diff
@@ -153,70 +153,77 @@ func generateStructs(_ events: [String : AnyObject]) throws -> NSString {
+            structString = replacePlaceholder(structString, placeholder: "<*!\(JETemplatePlaceholder.eventName.rawValue)*>", value: swiftyClassName(for: eventName)) //<*!event_name*> = Example
+            structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.eventName.rawValue)*>", value: eventName) //<*event_name*> = example
+            
+            let keys: [String] = (eventDic[JEPlistKey.payload.rawValue] as? [String]) ?? []
+            
+            /*
+             <*event_keyValueChain*> = kKey1 : key1, kKey2 : key2
+             */
+            let eventKeyValueChain: String = generateEventKeyValueChain(keys)
+            structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.keyValueChain.rawValue)*>", value: eventKeyValueChain)
+            
+            /*
+             <*event_cs_trackers_str*> = "console", "GA"
+             */
+            let eventCsTrackers: String = generateEventCsTrackers(eventDic[JEPlistKey.trackrs.rawValue] as! [String])
+            structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.eventTrackers.rawValue)*>", value: eventCsTrackers)
+            
+            /*
+             <*event_keysNames*> =
+             private let kKey1 = "key1"
+             private let kKey2 = "key2"
+             */
+            let eventKeysNames: String = try generateEventKeysNames(keys)
+            structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.keysNames.rawValue)*>", value: eventKeysNames)
+            
+            /*
+             <*event_keysVars*> =
+             let key1 : String
+             let key2 : String
+             */
+            let eventKeysVars: String = try generateEventKeysVars(keys)
+            structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.keysVars.rawValue)*>", value: eventKeysVars)
+            
+            /*
+             <*event_init*> =
+             init(<*event_init_params*>) {
              super.init()
              <*event_init_assigns_list*>
-         }
-         
-         <*event_init_params*> = test1: String, test2: String, test3: String
-         <*event_init_assigns_list*> =
-         self.test1 = test1
-         self.test2 = test2
-         self.test3 = test3
-         */
-        let eventInit: String = try generateEventInit(keys)
-        structString = replacePlaceholder(structString, placeholder: "<*\(JETemplatePlaceholder.eventInit.rawValue)*>", value: eventInit)
-        
-        structsArray.append(structString)
+             }
+             
+             <*event_init_params*> = test1: String, test2: String, test3: String
+             <*event_init_assigns_list*> =
+             self.test1 = test1
+             self.test2 = test2
+             self.test3 = test3
+             */
+            var eventInit: String = try generateEventInit(keys)
```

Comment:
The code generated is pretty ugly in case of empty events:

```
@objc public class JEEventEmptyEvent:NSObject,JEEvent {

    //JEEvent protocol
    public let name: String = "EmptyEvent"

    public var payload: Payload {
        return [:]
    }

    public var registeredTrackers: [String] {
        return []
    }

    //keys
    

    

    
}
```

Please change the code generator accordingly.